templates.rst

```diff
@@ -0,0 +1,1061 @@
+                {% for item in navigation %}
+                    <li><a href="{{ item.href }}">{{ item.caption }}</a></li>
+                {% endfor %}
+            </ul>
+        </body>
+    </html>
+
+Twig syntax is based on these three constructs:
+
+* ``{{ ... }}``, used to display the content of a variable or the result of
+  evaluating an expression;
+* ``{% ... %}``, used to run some logic, such as a conditional or a loop;
+* ``{# ... #}``, used to add comments to the template (unlike HTML comments,
+  these comments are not included in the rendered page).
+
+You can't run PHP code inside Twig templates, but Twig provides utilities to
+run some logic in the templates. For example, **filters** modify contents before
+being rendered, like the ``upper`` filter to uppercase contents:
+
+.. code-block:: twig
+
+    {{ title|upper }}
+
+Twig comes with a long list of `tags`_, `filters`_ and `functions`_ that are
+available by default. In Symfony applications you can also use these
+:doc:`Twig filters and functions defined by Symfony </reference/twig_reference>`
+and you can :doc:`create your own Twig filters and functions </templating/twig_extension>`.
+
+Twig is fast in the ``prod`` :doc:`environment <configuration-environments>`
+(because templates are compiled into PHP and cached automatically), but
+convenient to use in the ``dev`` environment (because templates are recompiled
+automatically when you change them).
+
+Twig Configuration
+~~~~~~~~~~~~~~~~~~
+
+Twig defines several configuration options to define things like the format used
+to display numbers and dates, the template caching, etc. Read the
+:doc:`Twig configuration reference </reference/configuration/twig>` to learn about them.
+
+Creating Templates
+------------------
+
+Template Naming
+~~~~~~~~~~~~~~~
+
+Symfony recommends the following for template names:
+
+* Use `snake case`_ for filenames and directories (e.g. ``blog_posts.twig``,
+  ``admin/default_theme/blog/index.twig``, etc.);
+* Define two extensions for filenames (e.g. ``index.html.twig`` or
+  ``blog_posts.xml.twig``) being the first extension (``html``, ``xml``, etc.)
+  the final format that the template will generate.
+
+Although templates usually generate HTML contents, they can generate any
+text-based format. That's why the two-extension convention simplifies the way
+templates are created and rendered for multiple formats.
+
+For example, if the route includes the :ref:`special _format routing parameter <routing-format-parameter>`
+(e.g. ``/index.{_format}``) you can browse different URLs (``/index.html``,
+``/index.xml``, ``/index.json``) to render different templates.
+
+Although Symfony doesn't select the template automatically based on the request
+format, you can do that yourself. Instead of rendering the ``'index.html.twig'``
+template, if your application supports different formats, you can render
+``'index.'.$format.'.twig'``, where ``$format = $request->getRequestFormat()``.
```

Comment:
I think the whole format thing is not relevant anymore. At best, this is a small section that comes way later. I think almost nobody should use templates to render json, xml, etc anymore.